src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const states = [];
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => {
+          delete currentState[key];
+        });
+        break;
 
+      case 'clear':
+        currentState = {};
```

Comment:
The action of copying the currentState is repeated in all cases of the switch statement. According to the DRY principle, you should perform this action once after the switch statement to avoid code duplication. This will also ensure that any future actions added to the switch will not require adding another copy of the currentState.